### ISSUE TITLE
"missing display name" when using React.createClass

### ISSUE THREAD

**Issue by fedemp** [yannickcr/eslint-plugin-react#1407]
Hello people. I'm having trouble linting some old code from a project that triggers the "Component definition is missing display name (react/display-name)" warning.

Minimal example:

```js
import React from "react";

const component = React.createClass({
	displayName: "MyComponent",
	render: function() {
		return (<div>Hello</div>);
	}
});

export default component;
```

Used .eslintrc

```js
{
	"parserOptions": {
		"sourceType": "module",
		"ecmaFeatures": {
			"jsx": true
		}
	},

	"env": {
		"es6": true
	},

	"plugins": [
		"react",
	],

	"rules": {
		"react/display-name": [2],
	}
}
```

**Comment by jseminck** [comment]
I think you have to use:

```
    settings: {
      react: {
        createClass: 'createClass'
      }
    }
```

**Comment by ljharb** [comment]
The latter isn't necessary here; just the createClass pragma.

**Comment by fedemp** [comment]
@jseminck Your suggestion actually worked. Closing. :)

**Comment by salemhilal** [comment]
For people in the future, I ran into this same issue, and adding `createClass: 'createClass'` to `settings.react` did the trick.
